adsmp/app.py

```diff
@@ -541,18 +543,26 @@ def checksum(self, data, ignore_keys=('mtime', 'ctime', 'update_timestamp')):
         @return: checksum
         """
         assert isinstance(ignore_keys, tuple)
-        
+
         if isinstance(data, basestring):
-            return hex(zlib.crc32(unicode(data)) & 0xffffffff)
+            if sys.version_info > (3,):
+                data_str = data.encode('utf-8')
+            else:
+                data_str = unicode(data)
+            return hex(zlib.crc32(data_str) & 0xffffffff)
         else:
             data = deepcopy(data)
             # remove all the modification timestamps
-            for k,v in data.items():
+            for k, v in list(data.items()):
```

Comment:
No need to call list.

Reply:
List() (or something like it) is needed because we're deleting keys from the dictionary as we're iterating over the keys/values. We could also use a shallow copy or something but we can't reuse the exact syntax from Python 2.